services/app-graphql/src/schema.graphql

```diff
@@ -1221,6 +1236,98 @@ type Rate {
     revisions: [RateRevision!]!
 }
 
+"""
+ContractRevision is a single instance of all the contract specific data in a submission
+it contains the unlock and submit info.
+"""
+type ContractRevision {
+    id: ID!
+    createdAt: DateTime
+    updatedAt: DateTime
+    "submitInfo is the who/when/why for this revision being submitted. An Unlocked revision has no submitInfo"
+    submitInfo: UpdateInformation
+    "unlockInfo is the who/when/why for this revision being unlocked. A DRAFT or SUBMITTED revision will not have one"
+    unlockInfo: UpdateInformation
+
+    "formData is all the contract specific info part of this submission"
+    formData: ContractFormData
+}
+
+"SubmittableRevision is what can appear in a submission"
+union SubmittableRevision = ContractRevision | RateRevision
+
+"ContractEditionReason is a hint as to why this edition was created"
+enum EditionReason {
+    "A submission of this contract and zero or more newly created rates"
+    SUBMISSION
+    "A submission of one of the rates related to this contract"
+    RATE_SUBMISSION
+    "A submission of a rate related to this contract, removing the link between the two"
+    RATE_UNLINK
+    "A submission of an unrelated rate, linking it to this contract"
+    RATE_LINK
+}
+
+"""
+ContractEdition is a snapshot of a contract and all its related rates in time along with 
+information about what submission triggered the new edition.
+"""
+type ContractEdition {
+    "cause is a hint as to why this edition was created"
+    cause: EditionReason!
+    "submitInfo provides the submission reason/date/by for this edition"
+    submitInfo: UpdateInformation!
+    "submittedRevisions is a list of contract and/or rate revisions that were submitted to create this edition"
+    submittedRevisions: [SubmittableRevision!]!
```

Comment:
The reason I like "submitted" is that the revisions in that list were indeed submitted in order to create this entry. Every revision in submittedRevisions will have the same submitInfo as included above. So in the omission case, a related rate is unlocked, unlinked from this contract, then submitted, creating an entry in this list. 

The only time submittedRevisions can have more than one entry is on a CONTRACT_SUBMISSION, when a contract is resubmitted it can also submit fresh rates. So maybe we could have different types for a RATE_SUBMISSION where it would just be `submittedRateRevision` instead of `submittedRevisions`